### DISCUSSION TITLE
Record Exception does not happen out of the box.

### DISCUSSION THREAD

**Discussion by robvanpamel** [open-telemetry/opentelemetry-dotnet#4581] [Q&A]
Hi

It might look like a detail, but after I instrumented my AspNetCore application to record exception, this still didn't happend. 

`````
tracerProviderBuilder.AddAspNetCoreInstrumentation(options =>
{
     options.RecordException = true;
});
`````

I still have to add an asp.net ExceptionHandler where I record the exception, like mentioned below. This is not really the behavior that I expected, but it might be by design. 

````
        app.UseExceptionHandler(exceptionHandlerApp =>
        {
            exceptionHandlerApp.Run(async context =>
            {
                // Get the triggered exception 
                var exceptionHandler = context.Features.Get<IExceptionHandlerFeature>();
                var exception = exceptionHandler.Error;

                // track it inside OpenTelemetry
                Activity.Current.RecordException(exception);
            });
        });
````

Is this normal behavior, or am I doing something wrong? 

I am using .NET 6 LST.

**Comment by vishweshbankwar** [comment]
@robvanpamel - Are you expecting exceptions handled inside `app.UseExceptionHandler` to be reported when you do `options.RecordException = true`? If Yes, then the behavior you are observing is by design. Exceptions handled in Middleware are not reported automatically even when you do `options.RecordException = true`. Only Unhandled exceptions are reported.
